code/modules/integrated_electronics/core/printer.dm

```diff
@@ -215,14 +242,38 @@
 
 				if(program["requires_upgrades"] && !upgraded)
 					to_chat(usr, "<span class='warning'>This program uses unknown component designs. Printer upgrade is required to proceed.</span>")
-				else
+				else if(fast_clone)
 					var/datum/component/material_container/materials = GetComponent(/datum/component/material_container)
 					if(materials.use_amount_type(program["metal_cost"], MAT_METAL))
 						var/obj/item/assembly = SScircuit.load_electronic_assembly(get_turf(src), program)
 						to_chat(usr,  "<span class='notice'>[assembly] has been printed from the provided template!</span>")
 						playsound(src, 'sound/items/poster_being_created.ogg', 50, TRUE)
 					else
 						to_chat(usr, "<span class='warning'>You need [program["metal_cost"]] metal to build that!</span>")
+				else
+					var/datum/component/material_container/materials = GetComponent(/datum/component/material_container)
+					if(!materials.use_amount_type(program["metal_cost"], MAT_METAL))
+						to_chat(usr, "<span class='warning'>You need [program["metal_cost"]] metal to build that!</span>")
+						return
+					var/cloning_time = program["metal_cost"] / 150
+					cloning_time = min(cloning_time, MAX_CIRCUIT_CLONE_TIME)
+					cloning = TRUE
+					clone_countdown = cloning_time
+					to_chat(usr, "<span class='notice'>You begin printing a custom assembly. This will take approximately [round(cloning_time / 60, 0.1)] minute(s). You can still print \
+					off normal parts during this time.</span>")
+					playsound(get_turf(src), 'sound/items/poster_being_created.ogg', 50, TRUE)
```

Comment:
playsound(src, ...

Reply:
While I'm on the subject, when should I use one or the other? I've always generally just used `src` but nowadays I'm called out seemingly at random.